pkgs/development/libraries/science/networking/ns3/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, fetchFromGitHub, fetchurl
+, python
+
+# for binding generation (untested)
+, castxml ? null, pygccxml ? null
+
+# for documentation
+, doxygen ? null, graphviz ? null, imagemagick ? null
+
+# for manual, tetex is used to get the eps2pdf binary
+# texlive to get latexmk. building manual still fails though
+, dia, tetex ? null, ghostscript ? null, texlive ? null
+
+, withDoc ? false
+, withManual ? false
+, withGsl ? false
+
+# --enable-examples
+, withExamples ? false
+
+# --enable-tests
+, withTests ? true
+, generateBindings ? false
+
+# All modules can be enabled by choosing 'all_modules'.
+# included here the DCE mandatory ones
+, modules ? [ "core" "network" "internet" "point-to-point"]
+, pkgs
+, lib
+}:
+
+stdenv.mkDerivation rec {
+  name = "ns-3.${version}";
+  version = "27.0";
+
+  # the all in one fetches netanim etc.
+  # https://www.nsnam.org/release/ns-allinone-3.27.tar.bz2;
+  src = fetchFromGitHub {
+    owner  = "nsnam";
+    repo   = "ns-3-dev-git";
+    rev = "${name}";
+    sha256 = "1w25qazsi4n35wzpxrfc1pdrid43gan3166c0bif196apras5phd";
+  };
+
+  buildInputs = lib.optionals generateBindings [ pkgs.castxml pygccxml ]
+    ++ stdenv.lib.optional withDoc [ doxygen graphviz imagemagick python ]
+    ++ stdenv.lib.optional withManual [ python.pkgs.sphinx dia tetex ghostscript
+   texlive.combined.scheme-medium ];
+
+  propagatedBuildInputs = with python.pythonPackages; with pkgs; [ gcc6 python ]
+    ++ stdenv.lib.optional withGsl [ gsl-bin libgsl2 libgsl ]
```

Comment:
ns3 supports many optional libraries and configurations, i prefer to keep the build small rather than list all of them. I made an exception for this one because it's supposed to improve simulation speed but I wouldn't mind removing it altogether.

Reply:
If it increase speed, then I would enable it.

Reply:
I asked on the channel and in fact it seems to just increase the wifi precision according to https://www.nsnam.org/wiki/Installation#Installation and not necessarily increase speed. I fixed the package names (locally), just gsl was enough